vagen/envs/search/search_env.py

```
@@ -0,0 +1,326 @@
+    async def step(self, action_str: str) -> Tuple[Dict[str, Any], float, bool, Dict[str, Any]]:
+        print("action_str:", action_str)
+        self._steps_used += 1
+        reward = 0.0
+        done = False
+        info: Dict[str, Any] = {}
+
+        parsed = _parse_action(action_str)
+        info.update(parsed)
+
+        # Budget check
+        remaining_steps = max(0, self.config.max_steps - self._steps_used)
```

Ruff (0.15.2)

[error] Local variable `remaining_steps` is assigned to but never used

Remove assignment to unused variable `remaining_steps`

(F841)